tests/erc721/erc721_test.go

```diff
@@ -162,6 +166,105 @@ func TestOpenSeaProxyApproval(t *testing.T) {
 	}
 }
 
+func TestOpenSeaProxyPreApproval(t *testing.T) {
+
+	tests := []struct {
+		name            string
+		hasProxy        bool
+		wantPreApproved bool
+	}{
+		{
+			name:            "without proxy on mint",
+			hasProxy:        false,
+			wantPreApproved: false,
+		},
+		{
+			name:            "with proxy on mint",
+			hasProxy:        true,
+			wantPreApproved: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sim, nft := deploy(t)
+
+			if tt.hasProxy {
+				openseatest.SetProxyTB(t, sim, sim.Addr(tokenMinter), sim.Addr(proxy))
+			}
+
+			approvalEvents := make(chan *TestableERC721CommonEnumerableApprovalForAll)
+			nft.TestableERC721CommonEnumerableFilterer.WatchApprovalForAll(&bind.WatchOpts{}, approvalEvents, nil, nil)
+			defer close(approvalEvents)
+
+			testEvent := func(expected bool, approved bool) func(*testing.T) {
+				return func(t *testing.T) {
+					select {
+					case got := <-approvalEvents:
+						got.Raw = types.Log{}
```

Comment:
Use `cmpopts.IgnoreFields(TestableERC721CommonEnumerableApprovalForAll{}, "Raw")` and pass it to `ethtest.Comparers()` as an extra option.